tn generate doesn't add most ipad simulators

I'm running tn v4.2.0. When running `tn generate`, it successfully finds most of the XCode simulators, but doesn't add most recipes for iPad and iPad Pro. `tn generate` gives the following errors:

`[ERROR] Invalid recipe name: ipad-5th-generation-`
`[ERROR] Invalid recipe name: ipad-pro-9-7-inch-`
`[ERROR] Invalid recipe name: ipad-pro-12-9-inch-`
`[ERROR] Invalid recipe name: ipad-pro-12-9-inch-2nd-generation-`
`[ERROR] Invalid recipe name: ipad-pro-10-5-inch-`
`[ERROR] Invalid recipe name: ipad-6th-generation-`
`[ERROR] Invalid recipe name: ipad-pro-11-inch-`
`[ERROR] Invalid recipe name: ipad-pro-12-9-inch-3rd-generation-`
`[ERROR] Invalid recipe name: ipad-5th-generation-`
`[ERROR] Invalid recipe name: ipad-pro-9-7-inch-`
`[ERROR] Invalid recipe name: ipad-pro-12-9-inch-`
`[ERROR] Invalid recipe name: ipad-pro-12-9-inch-2nd-generation-`
`[ERROR] Invalid recipe name: ipad-pro-10-5-inch-`

I can still run the simulators by using the device-id, but it would be nice to have these shortcut recipes added automatically. These names seem to be coming from XCode. I'm running XCode v10.1.

--- a/lib/setup.js
+++ b/lib/setup.js
@@ -39,7 +39,7 @@ function generate() {
         if (config.android.emulators && config.android.emulators.length > 0) {
 
           config.android.emulators.forEach(function forEach(dev) {
-            var name = dev.name.toLowerCase().replace(/\.+/g, '').replace(/[^a-z0-9]+/g, '-');
+            var name = dev.name.toLowerCase().replace(/\.+/g, '').replace(/[^a-z0-9]+/g, '-').replace(/-$/,'');
             var recipe = ['--android', '--emulator', '--device-id', dev.name];
 
             if (!recipes.has(name) || !arraysIdentical(recipes.get(name), recipe)) {
@@ -51,7 +51,7 @@ function generate() {
         if (config.android.devices && config.android.devices.length > 0) {
 
           config.android.devices.forEach(function forEach(dev) {
-            var name = dev.name.toLowerCase().replace(/\.+/g, '').replace(/[^a-z0-9]+/g, '-');
+            var name = dev.name.toLowerCase().replace(/\.+/g, '').replace(/[^a-z0-9]+/g, '-').replace(/-$/,'');
             var recipe = ['--android', '--device', '--device-id', dev.id];
 
             if (!recipes.has(name) || !arraysIdentical(recipes.get(name), recipe)) {
@@ -71,7 +71,7 @@ function generate() {
               return;
             }
 
-            var name = dev.name.toLowerCase().replace(/[^a-z0-9]+/g, '-');
+            var name = dev.name.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/-$/,'');
             var recipe = ['--ios', '--device', '--device-id', dev.udid];
 
             if (!recipes.has(name) || !arraysIdentical(recipes.get(name), recipe)) {
@@ -97,7 +97,7 @@ function generate() {
           versions.forEach(function forEach(version) {
 
             iosSims[version].forEach(function forEach(dev) {
-              var name = dev.name.toLowerCase().replace(/[^a-z0-9]+/g, '-');
+              var name = dev.name.toLowerCase().replace(/[^a-z0-9]+/g, '-').replace(/-$/,'');
               var recipe = ['--ios', '--simulator', '--device-id', dev.udid];
 
               if (recipes.has(name) && versions[0] !== version) {

